rotkehlchen/balances/historical.py

```diff
@@ -94,68 +121,318 @@ def get_assets_amounts(
+            AND ph1.to_asset = ?
+            AND ph1.timestamp BETWEEN ? AND ?
+            AND ph2.from_asset IS NULL;
+        """.format(','.join('?' * len(assets)))
+        bindings = [
+            timestamp,
+            timestamp,
+            *(asset.identifier for asset in assets),
+            main_currency.identifier,
+            timestamp - DAY_IN_SECONDS,
+            timestamp + DAY_IN_SECONDS,
+        ]
+
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            found_prices = {row[0]: FVal(row[1]) for row in cursor.execute(querystr, bindings)}
+            for asset in assets:
+                if asset.identifier in found_prices:
+                    prices[asset] = found_prices[asset.identifier]
+                elif asset == main_currency:
+                    prices[asset] = ONE
+                else:
+                    missing_prices.append((asset.identifier, timestamp))
+
+        return prices, missing_prices
+
+    @staticmethod
+    def _update_balances(
+            event: HistoryEvent | Trade,
+            current_balances: dict[Asset, FVal],
+    ) -> str | None:
+        """Updates current balances for a trade or history event, checking for negative balances.
+        Zero balance assets are removed to avoid accumulating empty entries.
+
+        For trades:
+        - BUY means we receive base_asset and spend quote_asset
+        - SELL means we spend base_asset and receive quote_asset
+
+        Returns the event_identifier if a negative balance would occur, else None.
+        """
+        if isinstance(event, Trade):
+            if event.trade_type in {TradeType.BUY, TradeType.SETTLEMENT_BUY}:
+                spend_asset = event.quote_asset
+                receive_asset = event.base_asset
+                spend_amount = event.amount * event.rate
+                receive_amount = event.amount
+            else:
+                spend_asset = event.base_asset
+                receive_asset = event.quote_asset
+                spend_amount = event.amount
+                receive_amount = event.amount * event.rate  # type: ignore[assignment]  # AssetAmount is the same as fval
+
+            if current_balances[spend_asset] - spend_amount < ZERO:
+                return event.identifier
+
+            current_balances[spend_asset] -= spend_amount
+            current_balances[receive_asset] += receive_amount
+            if current_balances[spend_asset] == ZERO:
+                del current_balances[spend_asset]
+
+            return None
+
+        # history event
+        if (
+            (direction := event.maybe_get_direction()) is None or
+            direction == EventDirection.NEUTRAL
+        ):
+            return None
+
+        if direction == EventDirection.IN:
+            current_balances[event.asset] += event.balance.amount
+        else:
+            if current_balances[event.asset] - event.balance.amount < ZERO:
+                return event.event_identifier
```

Comment:
So I just realized it's not `event.identifier` but it's `event.event_identifier`.

Can you make sure that this is what frontend will need to link it?

We need to change this naming. It's really confusing. One is for the group (the one you use) and the other for the single individual event.

i would guess you would need for the single event no?